pkg/controller/provider/container/openstack/model.go

```diff
@@ -732,11 +734,36 @@ func (r *VolumeAdapter) GetUpdates(ctx *Context) (updates []Updater, err error)
 					}
 					return
 				}
+
 				if !volume.updatedAfter(m) {
 					return
 				}
+
 				volume.ApplyTo(m)
 				err = tx.Update(m)
+				if err == nil {
+					// If an attached volume has changed, we have to update the relevant VM revision
+					// to make sure it is revalidated.
+					klog.Info("Volume changed, updating attached VMs", "volume", volume.ID)
+					for _, attachment := range volume.Attachments {
+						vmID := attachment.ServerID
+						vm := &model.VM{
+							Base: model.Base{ID: vmID},
+						}
+						err = tx.Get(vm)
+						if err != nil {
+							klog.Info("VM not found, skipping", "vmID", vmID)
+							continue
+						}
+						vm.Revision += 1
```

Comment:
don't we risk missing an update this way? I mean, if the vm is also updated in openstack we'll get that the revision didn't change - maybe we should decrease the revision or set it to zero?

Reply:
I can set vm.RevisionValidated to 0 instead